reference/hoon-expressions/rune/bar.md

```diff
@@ -8,12 +8,77 @@ Core expressions produce cores. A core is a cell of `[battery payload]`.
 The `battery` is code, a battery of Nock formulas.  The `payload` is the data
 needed to run those formulas correctly.
 
-Four core runes (`|=`, `|.`, `|-`, and `|*`) produce a
+Five core runes (`|=`, `|.`, `|-`, `|*`, and `|$`) produce a
 core with a single arm, named `$`. We can recompute this arm
-with changes, which useful for recursion among other things:
+with changes, which is useful for recursion among other things.
 
 ## Runes
 
+### |$ "barbuc"
+
+Produces a wet gate that creates a structure parameterized by molds.
+
+##### Syntax
+
+Regular:
+```hoon
+|$  a=(lest term)  b=spec
+```
+
+AST:
+```hoon
+[%brbs sample=(lest term) body=spec]
+```
+
+##### Semantics
+
+`|$` is used to create a wet gate that is polymorphic in its input types and
+produces a structure. `a` is a `lest` of `term` used as identifiers for the
+input types. `b` is a structure built from elements of `a`. The output of `|$`
+is a `spec` obtained by substituting the input types parameterized by `a` into
+the `b`.
+
+##### Expands to
+
+I'm having some trouble deciphering this, would appreciate help. From `hoon.hoon`:
+```hoon
+ {$brbs *}  =-  ?~  -  !!
+                       [%brtr [%bscl -] [%ktcl body.gen]]
+                   %+  turn  `(list term)`sample.gen
+                   |=  =term
+                   ^-  spec
+                   =/  tar  [%base %noun]
+                   [%bsts term [%bssg tar [%bshp tar tar]]]
+```
```

Comment:
Expansion is as follows:

```hoon
|$  [a b]
hoon
```

becomes

```hoon
|*  [a=$-(* *) b=$-(* *)]
^:
hoon
```

(I leave out the `$~` around the `$-`, because I don't think they actually change anything? Not sure why they're there... cc @joemfb)

Reply:
The difference between `$-(* *)` and `$~(* $-(* *))` is material -- I invite you try it out!

Reply:
Oh, I had tried everything, except this:

```hoon
> (($-(* *) 5) 6)
0
> (($~(* $-(* *)) 5) 6)
6
```

Amended my expansion above. Can't say I exactly understand this though...

Reply:
Thanks https://github.com/urbit/docs/pull/877/commits/6bcae1514962d02d246cb198e67fcdc90d94d153

I need to look up how the most basic runes are defined in this fashion so I can figure out what the hell is going on here :)